### ISSUE TITLE
Nightscout : missing BGs

### ISSUE THREAD

**Issue by robertrub** [nightscout/AndroidAPS#2938]
Not sure if it is a NS or AAPS bug... Using last Dev for both.
In AAPS, NS is set to update only when charging. Usually, when charging the last BGs are filled up. They are no more filled. There are holes in the BGs on NS 

![Screenshot_20231018_131759_DuckDuckGo.jpg](https://github.com/nightscout/AndroidAPS/assets/76626474/f4989fee-8459-4e26-b2fa-ca1e9ec5e77e)

![Screenshot_20231018_132152_AAPS.jpg](https://github.com/nightscout/AndroidAPS/assets/76626474/069059de-27a5-4b2c-b338-a7ea34364f9d)

Trying a full resync of NS in AAPS. 350k entries to update...

**Comment by robertrub** [comment]
I see this alert on my NS. I don't know if it's related and why there was this alert.

![Screenshot_20231018_134133_DuckDuckGo.jpg](https://github.com/nightscout/AndroidAPS/assets/76626474/bd2b8c4f-fa39-4140-8893-6335615ba3dc)

Full resync is uploading ok for the moment.

**Comment by MilosKozak** [comment]
without logs i cannot resolve anything

**Comment by robertrub** [comment]
After a full resync, the holes are there.

Sorry for the logs. Here they are

![AndroidAPS_LOG_1697634941992.log.zip](https://github.com/nightscout/AndroidAPS/files/13025621/AndroidAPS_LOG_1697634941992.log.zip)

You have the approximate times on the graph.

**Comment by robertrub** [comment]
It looks like AAPS is uploading only during _in charge_ but doesn't fill up the past BGs.

Setting is G6 to xDrip to AAPS. AAPS does the BG upload to NS.

**Comment by robertrub** [comment]
More info : When I set xDrip to upload to NS and to fill in _the past 48h_, all the holes are filled up.

**Comment by MilosKozak** [comment]
I need logs 8-10AM

**Comment by robertrub** [comment]
Sorry, bizarre but I have a huge hole in the logs too. I'll recreate the situation and will come with newer logs.

**Comment by MilosKozak** [comment]
if you are using v3 nothing else using v1( xdrip) must upload to NS

**Comment by robertrub** [comment]
Till the "bug" only AAPS was uploading to NS and via API V3.
I used xDrip and v1, just to fill in the holes

**Comment by MilosKozak** [comment]
Around 17.20 there are upload errors caused by using xdrip uploads

**Comment by robertrub** [comment]
Yes possible as I had used xDrip just before testing to fill in the holes.

Just tested now again (no xDrip connection for hours) but still not uploading.

Will build again and test tomorrow with your last commit on NSCV3 start.

**Comment by robertrub** [comment]
@MilosKozak Sorry but your last commit did not solve the problem of filling the past BGs. No xDrip/V1 connection for over 18 hours.

![Screenshot_20231021_100101_Firefox.jpg](https://github.com/nightscout/AndroidAPS/assets/76626474/8809fd6f-4f45-45c4-be33-a625e71e5984)

Look around 9:56 when I reconnected the phone to the charger.

![AndroidAPS_LOG_1697875146048.log.zip](https://github.com/nightscout/AndroidAPS/files/13060226/AndroidAPS_LOG_1697875146048.log.zip)

Seems like this time, there are no more the connection errors alerts I had on my NS page. So the commit solved one problem for sure 😃

**Comment by MilosKozak** [comment]
i see data properly uploaded to NS
but i see errors like HTTP FAILED: java.io.IOException: unexpected end of stream on ...
which are related to your NS instance

**Comment by robertrub** [comment]
I see that NS Master reverted and updated to 15.0.1 Will update my NS and check again.
BBS

**Comment by robertrub** [comment]
Ok, I'm back to NS v15.0.2 Dev.

Good news : the missing BG backfill was a NS bug. Now, when on charge, AAPS is backfilling the BGs. There are no more holes.

The less good news : I got the _Failed authentication error_ on my NS page when AAPS connected :( But the data got updated anyway. I don't know if thid is a NS or AAPS bug... 

Maybe we can close this issue and keep an eye on the _Failed auth error_ problem later in another issue.
